src/WebAppDIRAC/WebApp/handler/VMDiracHandler.py

```diff
@@ -0,0 +1,238 @@
+            v = [str(f) for f in v]
+          condDict[str(k).replace("_", ".")] = v
+    except Exception:
+      raise
+    try:
+      if 'statusSelector' in self.request.arguments:
+        condDict['inst.Status'] = [str(self.get_argument('statusSelector'))]
+    except Exception:
+      pass
+
+    result = VMClient().getInstancesContent(condDict, sort, start, limit)
+    if not result['OK']:
+      callback = {"success": "false", "error": result["Message"]}
+      self.write(callback)
+      return
+    svcData = result['Value']
+    data = {'numRecords': svcData['TotalRecords'], 'instances': []}
+    for record in svcData['Records']:
+      rD = {}
+      for iP in range(len(svcData['ParameterNames'])):
+        param = svcData['ParameterNames'][iP].replace(".", "_")
+        if param == 'inst_LastUpdate':
+          rD[param] = record[iP].strftime("%Y-%m-%d %H:%M:%S")
+        else:
+          rD[param] = record[iP]
+      data['instances'].append(rD)
+    callback = {"success": "true", "result": data['instances'], "total": data['numRecords'], "date": None}
+    self.write(callback)
+
+  def web_stopInstances(self):
+    webIds = json.loads(self.get_argument('idList'))
+    result = VMClient().declareInstancesStopping(webIds)
+    callback = {"success": "true", "result": result}
+    self.write(callback)
+
+  def web_getHistoryForInstance(self):
+    instanceID = int(self.get_argument('instanceID'))
+    result = VMClient().getHistoryForInstanceID(instanceID)
+    if not result['OK']:
+      return result
+    svcData = result['Value']
+    data = []
+    for record in svcData['Records']:
+      rD = {}
+      for iP in range(len(svcData['ParameterNames'])):
+        param = svcData['ParameterNames'][iP].replace(".", "_")
+        if param == 'Update':
+          rD[param] = record[iP].strftime("%Y-%m-%d %H:%M:%S")
+        else:
+          rD[param] = record[iP]
+      data.append(rD)
+    callback = {"success": "true", "result": data, "total": len(data)}
+    self.write(callback)
+
+  def web_checkVmWebOperation(self):
+    operation = str(self.get_argument('operation'))
+    result = VMClient().checkVmWebOperation(operation)
+    if not result['OK']:
+      callback = {"success": "false", "error": result['Message']}
+      return self.write(callback)
+    data = result['Value']
+    callback = {"success": "true", "data": data}
+    self.write(callback)
+
+  def web_getHistoryValues(self):
+    try:
+      dbVars = [str(f) for f in json.loads(self.get_argument('vars'))]
+    except Exception:
+      dbVars = ['Load', 'Jobs', 'TransferredFiles']
+    try:
+      timespan = int(self.get_argument('timespan'))
+    except Exception:
+      timespan = 86400
+    result = VMClient().getHistoryValues(3600, {}, dbVars, timespan)
+    if not result['OK']:
+      callback = {"success": "false", "error": result['Message']}
+      return self.write(callback)
+    svcData = result['Value']
+    data = []
+    olderThan = Time.toEpoch() - 400
```

Comment:
Since we just [learned](https://github.com/DIRACGrid/DIRAC/issues/5275) that this is local time, maybe we should change to UTC (used in several places below)? CC @sfayer @atsareg